server/controllers/authController.ts

```diff
@@ -0,0 +1,190 @@
+import jwt from "jsonwebtoken";
+import { Request, Response, NextFunction } from "express";
+import User from "../models/userModel";
+
+const signToken = (id: string) => {
+  return jwt.sign(
+    { id },
+    process.env.JWT_SECRET as string,
+    { expiresIn: process.env.JWT_EXPIRES_IN as string } as jwt.SignOptions
+  );
+};
+
+const createSendToken = (user: any, statusCode: number, res: Response) => {
+  const token = signToken(user._id.toString());
+
+  res.cookie("jwt", token, {
+    expires: new Date(
+      Date.now() +
+        parseInt(process.env.JWT_COOKIE_EXPIRES_IN as string) *
+          24 *
+          60 *
+          60 *
+          1000
+    ),
+    httpOnly: true,
+    sameSite: process.env.NODE_ENV !== "development" ? "none" : "lax",
+    secure: process.env.NODE_ENV !== "development",
+  });
+
+  (user as any).password = undefined;
+  (user as any).passwordConfirm = undefined;
+
+  res.status(statusCode).json({
+    status: "success",
+    token,
+    data: {
+      user,
+    },
+  });
+};
+
+export const signup = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const newUser = await User.create(req.body);
+    if (!newUser._id) {
+      res.status(400).json({
+        status: "error",
+        message: "User ID not found after creation.",
+      });
+      return;
+    }
+    createSendToken(newUser, 201, res);
+  } catch (err: any) {
```

Comment:
This looks more like a `500` error - for any "catch all" errors it's usually a good idea to return 5xx group of errors, as they are probably caused by misconfiguration/errors in the server (rest should be caught and handled with proper error codes)